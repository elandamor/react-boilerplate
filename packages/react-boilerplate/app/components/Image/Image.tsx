import React, { SFC } from 'react';
import classNames from 'classnames';
// Styles
import Wrapper from './styles';

/**
 * @render react
 * @name Image component
 * @description Image component.
 * @example
 * <Image />
 */

interface IProps {
  className?: string;
  src: string;
}

const Image: SFC<IProps> = ({ className, src, ...rest }) => (
  <Wrapper
    className={classNames('a-image', className)}
    data-src={src}
    src={src}
    {...rest}
  />
);

Image.defaultProps = {
  className: '',
  src: '',
};

export default Image;
