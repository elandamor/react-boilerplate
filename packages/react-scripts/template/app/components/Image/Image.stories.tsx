import React from 'react';
import { storiesOf } from '@storybook/react';
import { Share } from 'react-feather';

import Button from './index';

storiesOf('Button', module)
.add('default', () => <Button />)
.add('default with icon', () => <Button icon={<Share />} />)
.add('outlined', () => <Button outlined />)
.add('outlined with icon', () => <Button outlined icon={<Share />} />)
.add('raised', () => <Button raised />)
.add('raised with icon', () => <Button raised icon={<Share />} />)

