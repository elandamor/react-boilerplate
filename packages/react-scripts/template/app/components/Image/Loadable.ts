/**
 *
 * Asynchronously loads the component for Image
 *
 */

import React from 'react';

// @ts-ignore
export default React.lazy(() => import('./index'));
