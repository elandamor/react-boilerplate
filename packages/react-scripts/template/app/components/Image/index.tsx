import React, { FC } from 'react';
import classNames from 'classnames';
// Styles
import Wrapper from './styles';

interface IProps {
  as?: string;
  className?: string;
  src: string;
}

/**
 * @render react
 * @name Image component
 * @description Image component.
 * @example
 * <Image src="image.png" />
 */

const Image: FC<IProps> = ({ as: T, className, src, ...rest }) =>
  T ? (
    // @ts-ignore
    <T
      className={classNames('a-image -as', className)}
      style={{ backgroundImage: `url(${src})` }}
      {...rest}
    />
  ) : (
    <Wrapper className={classNames('a-image', className)} src={src} {...rest} />
  );

export default Image;
