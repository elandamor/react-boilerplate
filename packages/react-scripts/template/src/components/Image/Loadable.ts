/**
 *
 * Asynchronously loads the component for Image
 *
 */

import React from 'react';

export default React.lazy(() => import('./index'));
