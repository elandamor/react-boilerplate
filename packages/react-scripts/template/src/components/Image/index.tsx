import React, { FC } from 'react';
import classNames from 'classnames';
// Styles
import Wrapper from './styles';

export interface IImageProps {
  as?: any;
  className?: string;
  src: string;
}

/**
 * @render react
 * @name Image component
 * @description Image component.
 * @example
 * <Image src="image.png" />
 */

const Image: FC<IImageProps> = ({ className, ...rest }) => (
  <Wrapper
    className={classNames('a-image', className)}
    {...rest}
  />
);

Image.defaultProps = {
  as: 'img',
}

export default Image;
