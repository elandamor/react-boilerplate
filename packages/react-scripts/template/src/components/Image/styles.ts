import styled, { keyframes, css } from 'styled-components';
import { IImageProps } from './index';

const fadeIn = keyframes`
  from {
    opacity: 0;
  }
`;

const Wrapper = styled.img<IImageProps>`
  background-color: #e8e8e8;
  height: auto;
  width: 100%;

  &.-lazy {
    background-position: center;
    background-repeat: no-repeat;
    background-size: cover;
    border: none;
    height: auto;
    visibility: hidden;
    width: 100%;
  }

  &.-lazy--loaded {
    animation-duration: 3s;
    animation-name: ${fadeIn};
    visibility: visible;
  }

  ${(props) => props.as && css`
    background-image: url(${props.src});
    background-position: center;
    background-repeat: no-repeat;
    background-size: cover;
  `}
`;

export default Wrapper;
