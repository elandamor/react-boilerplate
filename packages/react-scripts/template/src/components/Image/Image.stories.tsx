import React from 'react';
import { storiesOf } from '@storybook/react';

import Image from './index';

storiesOf('Image', module)
.add('default', () => <Image src="../images" />)

