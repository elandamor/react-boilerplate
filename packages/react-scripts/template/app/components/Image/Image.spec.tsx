// Image.spec.tsx
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import 'jest-styled-components';

import Image from './index';

describe('<Image />', () => {
  it('should display an image', () => {
    const component = renderer
      .create(<Image src="../../images/icon-512x512.png" />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
