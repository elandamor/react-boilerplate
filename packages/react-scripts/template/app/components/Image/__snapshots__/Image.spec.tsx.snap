// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> should display an image 1`] = `
.c0 {
  background-color: #e8e8e8;
  height: auto;
  width: 100%;
}

.c0.-lazy {
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  border: none;
  height: auto;
  visibility: hidden;
  width: 100%;
}

.c0.-lazy--loaded {
  -webkit-animation-duration: 3s;
  animation-duration: 3s;
  -webkit-animation-name: kOCoaO;
  animation-name: kOCoaO;
  visibility: visible;
}

<img
  className="a-image c0"
  data-src="../../images/icon-512x512.png"
  src="../../images/icon-512x512.png"
/>
`;
